### PR TITLE
Populate full variable and use ns time

### PR DESCRIPTION
**Pull request by Affie**

### CODE REVIEW

File: src/ApproxConv.jl [hunk 1]
```diff
@@ -103,6 +103,8 @@ function prepareCommonConvWrapper!(ccwl::CommonConvWrapper{T},
     ccwl.cpt[i].p = collect(1:size(ccwl.cpt[i].X,1)) # collect(1:length(ccwl.cpt[i].Y))
     ccwl.cpt[i].Y = zeros(ccwl.xDim)  # zeros(ccwl.partial ? length(ccwl.usrfnc!.partial) : ccwl.xDim )
     ccwl.cpt[i].res = zeros(ccwl.xDim) # used in ccw functor for AbstractRelativeFactorMinimize
+    # TODO JT - Confirm it should be updated here.
+    ccwl.cpt[i].factormetadata.fullvariables = copy(Xi)
```
Comment (dehann) [inline src/ApproxConv.jl:107]: is a copy necessary?  Factor operations should not be allowed to change the variable metadata?

File: src/FactorGraph.jl [hunk 2]
```diff
@@ -398,19 +398,32 @@ function addVariable!(dfg::AbstractDFG,
...
                       checkduplicates::Bool=true,
                       initsolvekeys::Vector{Symbol}=getSolverParams(dfg).algorithms)::DFGVariable
   #
+
+  if :ut in fieldnames(typeof(softtype))
+    Base.depwarn("Field `ut` (microseconds) for softtype $(softtype) has been deprecated please use DFGVariable.nstime, kwarg: nanosecondtime", :addVariable!)
+    if isnothing(nanosecondtime)
+      nanosecondtime = Nanosecond(softtype.ut*1000)
+    else 
+      @warn "Nanosecond time has been speciefied as $nanosecondtime, ignoring `ut` field value: $(softtype.ut)."
```
Comment (dehann) [inline src/FactorGraph.jl:415]: speciefied -> specified

File: src/ApproxConv.jl (hunk 1 above)
Comment (Affie) [reply src/ApproxConv.jl:107]: Copy was to be on the save side. I want to be sure the container does not change. It looks like it doesn't, but there are some comments in the code about constructing `Xi`:
> # TODO -- this build up of Xi is excessive and could happen at addFactor time
Perhaps `fullvairables` can be populated at construction time?

File: src/FactorGraph.jl (hunk 2 above)
Comment (Affie) [reply src/FactorGraph.jl:415]: thanks, missed that

File: src/ApproxConv.jl (hunk 1 above)
Comment (Affie) [reply src/ApproxConv.jl:107]: @dehann perhaps rather in `prepgenericconvolution`? Or will it not make a difference.

File: src/FactorGraph.jl [hunk 3]
```diff
@@ -608,6 +621,8 @@ function prepgenericconvolution(
         )
   #
   for i in 1:Threads.nthreads()
+    # TODO JT - Confirm it should be updated here. Also testing in prepareCommonConvWrapper!
+    ccw.cpt[i].factormetadata.fullvariables = copy(Xi)
```
Comment (dehann) [inline src/FactorGraph.jl:625]: again, not sure if the copy is really needed?

**Pull request merged**